File: Projects/Framework-Samples/PLINQO/VisualBasic/Tracker/Source/Tracker.Tests/QueryTests/ByEnumTests.vb
Imports System.Collections.Generic
Imports System.Linq
Imports NUnit.Framework
Imports Tracker.Core.Data

Namespace Tracker.Tests.QueryTests
    <TestFixture()> _
    Public Class ByEnumTests
        Inherits TestBase
        <Test()> _
        Public Sub ByTest()
            Using db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }
                Dim a As List(Of Task) = db.Task.ByStatus(Status.InProgress).ToList()
                Dim b As List(Of Task) = db.Task.ByStatus(Status.NotStarted).ToList()

                Dim d As List(Of Task) = db.Task.ByStatus(Status.InProgress, Status.NotStarted).ToList()
                Assert.AreEqual(a.Count + b.Count, d.Count)
            End Using
        End Sub

        <Test()> _
        Public Sub ByNullableTest()
            Using db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }
                Dim a As List(Of Task) = db.Task.ByPriority(CType(Nothing, System.Nullable(Of Priority))).ToList()
                Dim b As List(Of Task) = db.Task.ByPriority(Priority.Normal).ToList()
                Dim c As List(Of Task) = db.Task.ByPriority(Priority.High).ToList()

                Dim d As List(Of Task) = db.Task.ByPriority(Priority.Normal, CType(Nothing, System.Nullable(Of Priority))).ToList()
                Assert.AreEqual(a.Count + b.Count, d.Count)

                Dim e As List(Of Task) = db.Task.ByPriority(Priority.Normal, CType(Nothing, System.Nullable(Of Priority)), Priority.High).ToList()
                Assert.AreEqual(a.Count + b.Count + c.Count, e.Count)
            End Using
        End Sub
    End Class
End Namespace

File: Framework-Samples/PLINQO/CSharp/Tracker/Source/Tracker.Tests/QueryTests/ByEnumTests.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CodeSmith.Data.Linq;
using NUnit.Framework;
using Tracker.Core.Data;

namespace Tracker.Tests.QueryTests
{
    [TestFixture]
    public class ByEnumTests : TestBase
    {
        [Test]
        public void ByTest()
        {
                using (var db = new TrackerDataContext {Log = Console.Out})
                {
                    var a = db.Task.ByStatus(Status.InProgress).ToList();
                    var b = db.Task.ByStatus(Status.NotStarted).ToList();

                    var c = db.Task.ByStatus(Status.InProgress, null).ToList();
                    Assert.AreEqual(a.Count, c.Count);

                    var d = db.Task.ByStatus(Status.InProgress, Status.NotStarted).ToList();
                    Assert.AreEqual(a.Count + b.Count, d.Count);
                }

        }

        [Test]
        public void ByNullableTest()
        {
                using (var db = new TrackerDataContext {Log = Console.Out})
                {
                    var a = db.Task.ByPriority((Priority?)null).ToList();
                    var b = db.Task.ByPriority(Priority.Normal).ToList();
                    var c = db.Task.ByPriority(Priority.High).ToList();

                    var d = db.Task.ByPriority(Priority.Normal, null).ToList();
                    Assert.AreEqual(a.Count + b.Count, d.Count);

                    var e = db.Task.ByPriority(Priority.Normal, null, Priority.High).ToList();
                    Assert.AreEqual(a.Count + b.Count + c.Count, e.Count);

                    var f = db.Task.ByPriority(ComparisonOperator.NotEquals, null).ToList();
                    var g = db.Task.ByPriority(ComparisonOperator.Equals, null).ToList();

                }

        }
    }
}
